util/stringutil/string_util.go

```diff
@@ -168,18 +168,6 @@ func CompilePatternInner(pattern string, escape byte) (patWeights []rune, patTyp
 			if i < lenRunes-1 {
 				i++
 				r = runes[i]
-				if r == escapeRune || r == '_' || r == '%' {
-					// Valid escape.
-				} else {
```

Comment:
Why do you delete it? How does it work?

Reply:
before this PR, we compile the pattern
```
\% -> %
% -> any
a -> a
\a -> \a
```
after
```
\% -> %
% -> any
a -> a
\a -> a
```

so that SQL `select 'a' like '\\a'` is true.